src/landing-page/create-modal.js

```diff
@@ -0,0 +1,165 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useState } from '@wordpress/element';
+import {
+	// eslint-disable-next-line @wordpress/no-unsafe-wp-apis
+	__experimentalHStack as HStack,
+	// eslint-disable-next-line @wordpress/no-unsafe-wp-apis
+	__experimentalVStack as VStack,
+	// eslint-disable-next-line @wordpress/no-unsafe-wp-apis
+	__experimentalText as Text,
+	Modal,
+	Button,
+	TextControl,
+	TextareaControl,
+} from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import {
+	createBlankTheme,
+	createClonedTheme,
+	createChildTheme,
+} from '../resolvers';
+
+export const CreateThemeModal = ( { onRequestClose, creationType } ) => {
+	const [ errorMessage, setErrorMessage ] = useState( null );
+
+	const [ theme, setTheme ] = useState( {
+		name: '',
+		description: '',
+		author: '',
+	} );
+
+	const renderCreateButtonText = ( type ) => {
+		switch ( type ) {
+			case 'blank':
+				return __(
+					'Create and Activate Blank Theme',
+					'create-block-theme'
+				);
+			case 'clone':
+				return __( 'Clone Block Theme', 'create-block-theme' );
+			case 'child':
+				return __( 'Create Child Theme', 'create-block-theme' );
+		}
+	};
+
+	const createBlockTheme = async () => {
+		let constructionFunction = null;
+		switch ( creationType ) {
+			case 'blank':
+				constructionFunction = createBlankTheme;
+				break;
+			case 'clone':
+				constructionFunction = createClonedTheme;
+				break;
+			case 'child':
+				constructionFunction = createChildTheme;
+				break;
+		}
+
+		if ( ! constructionFunction ) {
+			return;
+		}
+		constructionFunction( theme )
+			.then( () => {
+				// eslint-disable-next-line no-alert
+				window.alert(
+					__(
+						'Theme created successfully. The editor will now load.',
+						'create-block-theme'
+					)
+				);
+				window.location = '/wp-admin/site-editor.php?canvas=edit';
+			} )
+			.catch( ( error ) => {
+				setErrorMessage(
+					error.message ||
+						__(
+							'An error occurred while attempting to create the theme.',
+							'create-block-theme'
+						)
+				);
+			} );
+	};
+
+	if ( errorMessage ) {
+		return (
+			<Modal
+				title={ __( 'Create Block Theme', 'create-block-theme' ) }
+				onRequestClose={ onRequestClose }
+			>
+				<p>{ errorMessage }</p>
+			</Modal>
+		);
+	}
+
+	return (
+		<Modal
+			title={ __( 'Create Block Theme', 'create-block-theme' ) }
+			onRequestClose={ onRequestClose }
+		>
+			<VStack style={ { gap: '20px' } }>
```

Comment:
```suggestion
			<VStack spacing="5">
```

I think `spacing` prop is better than inline styles. This is equivalent to 20px.

Reply:
Ah, that's what I was looking for.  Thank you.